### PR TITLE
Allow SQLite without built tag

### PR DESCRIPTION
**Pull request by duckbrain**
Uses run-time check for SQLite being included as a driver to avoid a direct dependence. Allows using SQLite while omitting `-tags sqlite` from the build.

```
package main

import _ "github.com/mattn/go-sqlite3" // Load SQLite3 CGo driver

func main() {
	// main application code using Pop
}
```

The `sqlite` built tag now just includes `github.com/mattn/go-sqlite3` automatically. This works similarly to [`time/tzdata`](https://pkg.go.dev/time/tzdata).

### CODE REVIEW

File: dialect_sqlite_tag.go
Comment (aeneasr) [inline dialect_sqlite_tag.go:2]: Isn't this still requiring the `sqlite` tag to work? Or do you mean that you can now import this library, without go-sqlite3 included (unless `-tags sqlite` is specified) and then just choose to do this in your own app?

**Review by aeneasr — CHANGES_REQUESTED** [review]

File: dialect_sqlite_tag.go
Comment (duckbrain) [reply dialect_sqlite_tag.go:2]: The second option. As long as the driver is loaded, Pop will work with it.

This change preserves backwards-compatible behavior, so you can include the `sqlite` tag to include the driver.

**Comment by aeneasr** [comment]
Sorry for not merging this, could you resolve the conflict? I'll merge right after!

**Review by aeneasr — APPROVED** [review]

**Pull request merged**
